src/content-tags-drawer/ContentTagsTree.jsx

```diff
@@ -56,7 +55,7 @@ const ContentTagsTree = ({ tagsTree, removeTagHandler, editable }) => {
             level={level}
             lineage={updatedLineage}
             removeTagHandler={removeTagHandler}
-            editable={editable}
+            canDelete={tag[key].canDelete}
```

Comment:
I think we need the `canTagObject` from the parent here.

The `canDelete` is "Can I remove this Tag from this Taxonomy?", not "Can I remove this ObjectTag from this Object?"

Reply:
Please see this [comment](https://github.com/openedx/frontend-app-course-authoring/pull/787#discussion_r1452807956).

Reply:
Hi, @pomegranited!
This canDelete is now mapped to the ObjectTag.delete permission right?
If we opt to use the ObjectTag permissions (which is fine), I think we should use the canChangeObjectTag permission instead.

This is a bit confusing. The ObjectTag permissions name is misleading because it is not related to the ObjectTag model itself but with a Taxonomy + ObjectId pair (no reference to ObjectTag or Tag).

The backend tagging function (used here) only updates ObjectTag in batches. If we "remove" a tag from an object we actually pass an empty array to the update function and check the ObjectTag.can_change.
We do not use the ObjectTag.can_add and ObjectTag.can_delete permissions right now.

Today, both link to the same `can_change_object_tag`check (again, misleading name), but it can change in the future.

https://github.com/openedx/openedx-learning/blob/e94dcf7e4c5cb06c8fb3166b3af13be9fdd3d3f7/openedx_tagging/core/tagging/rules.py#L185-L187

If we use canDelete here we may be checking a different permission in the backend.

Reply:
@rpenido  Hmm.. At the risk of worrying about future use cases, I think using "can_change_objecttag" could be problematic in the future, when we might let users edit their free-text object tags?

I think a better solution is to change the `ObjectTagView.update` method to check "can_add" + "can_delete" permissions instead of "can_change" -- implementation-wise it does the same thing, but it's logically more accurate.

Please see https://github.com/openedx/openedx-learning/pull/138/commits/3e24b66222c457214e4a8b408678261b037a50ea